### ISSUE TITLE
`converter` type in attrs not correctly passed on to `__init__`

### ISSUE THREAD

**Issue by pvandyken** [microsoft/pyright#7713]
**Describe the bug**
In pyright 1.1.358, the generated `__init__` types for attrs-wrapped classes no longer respect functions or constructors provided to the `converter` argument of `attrs.field`. The now-expected type seems to vary depending on the converter, I haven't found an exact pattern. Simple converters like `str` seem to work fine.

**Code or Screenshots**
```python
# test.py
from __future__ import annotations

import attrs

@attrs.define
class A:
    a: dict[str, str] = attrs.field(converter=dict)


test = A(a={"foo": "bar"})
```

```bash
$ pyright test.py
.../test.py
  .../test.py:10:12 - error: Argument of type "dict[str, str]" cannot be assigned to parameter "a" of type "Iterable[list[str]] | Iterable[list[bytes]]" in function "__init__"
    Type "dict[str, str]" cannot be assigned to type "Iterable[list[str]] | Iterable[list[bytes]]"
      "dict[str, str]" is incompatible with "Iterable[list[str]]"
        Type parameter "_T_co@Iterable" is covariant, but "str" is not a subtype of "list[str]"
          "str" is incompatible with "list[str]"
      "dict[str, str]" is incompatible with "Iterable[list[bytes]]"
        Type parameter "_T_co@Iterable" is covariant, but "str" is not a subtype of "list[bytes]"
          "str" is incompatible with "list[bytes]" (reportArgumentType)
1 error, 0 warnings, 0 informations 
```

**VS Code extension or command-line**
Just the command-line, this version is not yet shipped with pylance.

**Comment by erictraut** [comment]
I've fixed the regression, but it's important to note that support for "converter" is experimental and is contingent on the acceptance of [PEP 712](https://peps.python.org/pep-0712/). The steering council recently signaled that they are disinclined to accept this PEP, although they haven't issued their final decision. I've disabled this functionality by default because I don't want pyright users to assume that it will be implemented. It is still available if you specify "enableExperimentalFeatures: true" in your pyright configuration file. This reflects the fact that the functionality is experimental and will be removed in the future if the steering council rejects PEP 712.

**Comment by erictraut** [comment]
This is addressed in pyright 1.1.364.
